### ISSUE TITLE
iPhone X/iOS swipe gesture - Keyboard from other apps resizes Capacitor app

### ISSUE THREAD

**Issue by benharpers** [ionic-team/capacitor#1873]
Description of the problem:

Affected platform
- [ ] Android
- [X] iOS
- [ ] electron
- [ ] web

OS of the development machine
- [ ] Windows
- [X] macOS
- [ ] linux

Other information:

Might be only on iPhone X and newer devices that have the bottom swipe gestures and the white bar at the bottom of the screen.

Capacitor version:1.1

node version: 9.4

npm version: 6.10.1

CocoaPods version: 1.7.4

Steps to reproduce:

If you switch from your Capacitor app to an app that has the keyboard open (such as Messages), when you switch back (specifically by swiping the bottom bar on iPhone X), the Capacitor app will be shrunk for a second as if it had the keyboard open, but with black at the bottom where the keyboard was, then it will resize itself. 

Link to sample project:

Let me know if you cannot reproduce

**Comment by benharpers** [comment]
Thanks for the update, not sure if it fixed the problem though. I copied just the Keyboard.m from the repo and seems to still happen. The `if` conditions you added aren't running while it's in the background, apparently the keyboard hide event from the other app like Messages happens after the Capacitor app becomes active, so `UIApplicationStateActive` is the state at that point.

I think this is the condition to check for:

```
  if (UIKeyboardIsLocalUserInfoKey.boolValue == true) {
...
  }
```

https://developer.apple.com/documentation/uikit/uiresponder/1621603-keyboardislocaluserinfokey

It appears to work, at least on my old iPhone X.

**Comment by jcesarmobile** [comment]
I first tried with UIKeyboardIsLocalUserInfoKey and it was returning true sometimes, but my app wasn't showing a keyboard, so I don't think that value it's trustworthy. Checking for `UIApplicationStateActive` worked fine for me when switching from Twitter app, didn't try with Messages app, will try again.

**Comment by benharpers** [comment]
Yeah I did see the app state method work sometimes also, and I might have noticed some unreliability with the keyboardIsLocal option... perhaps it would be a good idea to check that both are true.

**Comment by benharpers** [comment]
I think part of the reason it's happening specifically with the swipe gesture on iPhone X is that the app activate and keyboard close event happen almost at the same time with that gesture, maybe either condition could happen first and maybe that is where the unreliability comes from.

**Comment by jcesarmobile** [comment]
I've been able to reproduce, so I'm going to reopen.
I've checked with just `UIKeyboardIsLocalUserInfoKey` and a combination of `UIKeyboardIsLocalUserInfoKey` and `UIApplicationStateActive` and they don't fix it and can even cause that the black area is not removed.

**Comment by benharpers** [comment]
Thanks, sorry for the trouble! Not sure why Apple thought it was a good idea to let apps know about other apps keyboards closing!

**Comment by oliverandersencox** [comment]
@jcesarmobile I have this issue when using the in app browser for social logins, the keyboard will appear and disappear ad the browser closes. However after the black space is left and does not resize.

Is there a method I can use to resize the app again after this?

**Comment by Sasha4ever** [comment]
Are there any news/fixes?

**Comment by Sasha4ever** [comment]
Are there some news?
